src/Adapter/MSTest.TestAdapter/Execution/TestMethodInfo.cs

```diff
@@ -117,14 +118,9 @@ public virtual TestResult Invoke(object[] arguments)
             watch.Start();
             try
             {
-                if (IsTimeoutSet)
-                {
-                    result = ExecuteInternalWithTimeout(arguments);
-                }
-                else
-                {
-                    result = ExecuteInternal(arguments);
-                }
+                result = IsTimeoutSet
+                    ? ExecuteInternalWithTimeout(arguments)
+                    : ExecuteInternal(arguments, CancellationToken.None, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
Pls add a comment to explain why you need this blocking call

Reply:
Also are we sure we want block here? It means that starting from now we're pushing the thread pool in this hot loop(test invoke is our host loop I suppose) are the perf good? Cannot we move this wait above in the chain a bit?